test/quantization/core/test_docs.py

```diff
@@ -0,0 +1,96 @@
+# Owner(s): ["oncall: quantization"]
+
+import torch
+from torch.testing._internal.common_quantization import (
+    QuantizationTestCase,
+    SingleLayerLinearModel,
+)
+from os.path import exists
+
+
+class TestQuantizationDocs(QuantizationTestCase):
+    r"""
+    The tests in this section import code from the quantization docs and check that
+    they actually run. In cases where objects are undefined in the code snippet, they
+    must be provided in the test.
+    """
+
+    def _get_code(self, filename, first_line, last_line, offset=2, strict=True):
```

Comment:
nit: I wonder if there is a way to avoid using the line numbers? My concern is that any change in the RST doc would require change in all the test calls. One option is to use RST comments with a keyword.

Reply:
Yeah, I can make it a unique identifier, good idea! I considered detecting API example before but wanted each chunk to have a different test, with a unique identifier, that would solve that issue.